### ISSUE TITLE
_MSC_VER not defined

### ISSUE THREAD

**Issue by galeone** [adamrehn/ue4-docker#51]
Hi, thank you for this great project!

I've successfully built the docker images from strach, and until I was using inside my project code that doesn't depends on `_MSV_VER` everything was going fine.

But righ now, I've added to my project a third party library that is rich of

```
#if defined(_MSC_VER)
// do stuff
```

Unfortunately, the macro `_MSC_VER` is not defined when using the image "build-prerequisites" that contains Visual Studio 2017.

I've googled around, and all I found is this thread: https://developercommunity.visualstudio.com/content/problem/403959/-msc-ver-is-not-defined.html

In short: the problem is present when there are 2 versions of visual studio installed in the same machine (my guess, not sure if this is correct).

Looking at the file https://github.com/adamrehn/ue4-docker/blob/master/ue4docker/dockerfiles/ue4-build-prerequisites/windows/install-prerequisites.bat#L12 I see that you're using vs2019 to download and install the vs2017 components - maybe this can cause the issue?

Also, if you have the time to have a look and see if even using your locally built images you have the same issue (missing definition of the `_MSC_VER` macro) it would be great.

Thank you!

**Comment by adamrehn** [comment]
Hmmm, that's very strange. I'll build a fresh image and have a look. Are you encountering this issue just when building C++ code through UnrealBuildTool or also when invoking the `cl.exe` compiler directly?

**Comment by galeone** [comment]
I'm really sorry, this was a false alarm not related to the ue4-docker project!

My fault, It was a mess created by including a third-party library in unreal and not properly hiding the unreal types -> including the library -> restoring the unreal types.

There was some macro in the included code that was messing with everything and caused the problem. I fixed it yesterday and I forgot to update the issue.

**Comment by adamrehn** [comment]
No worries, I'm glad you got it sorted out in the end! The Unreal Engine headers are a bit notorious for not playing nicely with third-party library headers, so this sort of thing is a fairly common frustration when integrating third-party libraries. (The other major bugbear is UBT treating warnings as errors, which often necessitates manually disabling individual warnings using `#pragma` directives.)
